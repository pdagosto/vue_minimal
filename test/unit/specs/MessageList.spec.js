import { shallowMount } from '@vue/test-utils';
import MessageList from '@/components/MessageList';

describe('MessageList.test.js', () => {
    let cmp;

    beforeEach(() => {
        cmp = shallowMount(MessageList, {
            propsData: {
                messages: ['Cat', 'Dog', 'Parakeet']
            }
        });
    });

    it('has received ["Cat"] as the message property', () => {
        expect(cmp.vm.messages).toEqual(['Cat', 'Dog', 'Parakeet']);
    });

    it('has the expected html structure', () => {
        expect(cmp.element).toMatchSnapshot();
    });
});
