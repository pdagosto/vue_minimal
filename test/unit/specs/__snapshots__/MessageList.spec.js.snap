// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList.test.js has the expected html structure 1`] = `
<ul>
  <li>
    
        Cat
    
  </li>
  <li>
    
        Dog
    
  </li>
  <li>
    
        Parakeet
    
  </li>
</ul>
`;
